viewing_party/party.py

```diff
@@ -1,23 +1,124 @@
+    return user_data
+        
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+    sum_of_rating = 0
+    for item in user_data["watched"]:
+        for movie in item:
+            if movie == "rating":
+                sum_of_rating  += item[movie]
+                avg_rating = sum_of_rating /len(user_data["watched"])
+    return avg_rating
 
-
+def get_most_watched_genre(user_data):
+    genre_list =[]
+    for most_watched in user_data["watched"]:
+        for movie in most_watched:
+            if movie == "genre":
+                genre_list.append(most_watched[movie])
+    for genre in range(len(genre_list)):
+        genre_list.count(genre_list[genre])
+        most_watched_genre = genre_list[genre]
+        return most_watched_genre
+    else:
+        return None
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_list =[]
+    friends_movie_list=[]
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    for movie in user_data["watched"]:
+        if movie not in friends_movie_list:
+            unique_list.append(movie)
+    return unique_list
 
+def get_friends_unique_watched(user_data):
+    unique_friends_list =[]
+    user_movie_list=[]
+    for movie in user_data["watched"]:
+        if movie not in user_movie_list:
+            user_movie_list.append(movie)
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_movie_list and movie not in unique_friends_list:
+                unique_friends_list.append(movie)
+    return unique_friends_list
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
-
+def get_available_recs(user_data):
+    recommended_movie_list=[]
+    subscriptions_list=[]
+    unique_watched = get_friends_unique_watched(user_data)
+    
+    for user in user_data["subscriptions"]:
+            subscriptions_list.append(user)
+    for movie in unique_watched:
+        if movie["host"]in subscriptions_list:
+            recommended_movie_list.append(movie)
+    return recommended_movie_list
+        
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommended_list = []
+    most_watched_genre = get_most_watched_genre(user_data)
+    unique_watched = get_friends_unique_watched(user_data)
+    
+    for watch in user_data["watched"]:
+        if watch is None:
+            return recommended_list

```

Comment:
This for loop is redundant. We don't need to check to see if any of the individual movies are None.